### PR TITLE
Add r-fastmatch recipe

### PR DESCRIPTION
**Pull request by chambm**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by chambm** [comment]
@bgruening Any idea what's going wrong here? I'm trying to build it locally with simulate_travis but can't manage it. What's the recommended docker image to start from that has conda/gcc and supports docker-in-docker?

**Comment by bgruening** [comment]
Mh, this is the error: `bin/R: No such file or directory`
and the docker container which we are using is here: https://github.com/bioconda/bioconda-recipes/blob/master/config.yml#L8

The docker in docker stuff you don't need. This fail earlier and we don't do Docker in Docker, can tell you more via chat if you like.

**Comment by chambm** [comment]
Well, with --disable-docker, here are the commands I tried and the output. I tried with and without --dirty.
**(root) [root@4d77fd2a6b14 bioconda-recipes]# conda build recipes/r-fastmatch -n --dirty**
```
BUILD START: r-fastmatch-1.1_0-r3.3.2_0
    (actual version deferred until further download or env creation)
updating index in: /opt/conda/conda-bld/linux-64
updating index in: /opt/conda/conda-bld/noarch

The following NEW packages will be INSTALLED:

    bzip2:           1.0.6-1     conda-forge
    ca-certificates: 2017.1.23-1 conda-forge
    cairo:           1.14.6-4    conda-forge
    cloog:           0.18.0-0    defaults
    curl:            7.52.1-0    conda-forge
    fontconfig:      2.12.1-4    conda-forge
    freetype:        2.7-1       conda-forge
    gcc:             4.8.5-7     defaults
    gettext:         0.19.7-1    conda-forge
    glib:            2.51.4-0    conda-forge
    gmp:             6.1.2-0     conda-forge
    graphite2:       1.3.9-0     conda-forge
    gsl:             2.2.1-1     conda-forge
    harfbuzz:        1.4.3-0     conda-forge
    icu:             58.1-1      conda-forge
    isl:             0.12.2-0    defaults
    jpeg:            9b-0        conda-forge
    libffi:          3.2.1-3     conda-forge
    libgcc:          5.2.0-0     defaults
    libiconv:        1.14-4      conda-forge
    libpng:          1.6.28-0    conda-forge
    libtiff:         4.0.6-7     conda-forge
    libxml2:         2.9.4-4     conda-forge
    mpc:             1.0.3-4     conda-forge
    mpfr:            3.1.5-0     conda-forge
    ncurses:         5.9-10      conda-forge
    openssl:         1.0.2k-0    conda-forge
    pango:           1.40.4-0    conda-forge
    pcre:            8.39-0      conda-forge
    pixman:          0.34.0-0    conda-forge
    r-base:          3.3.2-5     conda-forge
    readline:        6.2-0       conda-forge
    tk:              8.5.19-1    conda-forge
    xz:              5.2.2-0     conda-forge
    zlib:            1.2.8-3     conda-forge

/opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/bin/.gcc-post-link.sh: line 10: lsb_release: command not found

WARNING: conda-build appears to be out of date. You have version 2.1.10 but the
latest version is 2.1.13. Run

conda update -n root conda-build

to get the latest version.

Source cache directory is: /opt/conda/conda-bld/src_cache
Found source in cache: fastmatch_1.1-0.tar.gz
Extracting download
BUILD START (revised): r-fastmatch-1.1_0-r3.3.2_0
Package: r-fastmatch-1.1_0-r3.3.2_0
source tree in: /opt/conda/conda-bld/r-fastmatch_1494873960049/work/fastmatch
+ source /opt/conda/bin/activate /opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p
+ mv DESCRIPTION DESCRIPTION.old
+ grep -v '^Priority: ' DESCRIPTION.old
+ /opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/bin/R CMD INSTALL --build .
* installing to library ‘/opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/R/library’
* installing *source* package ‘fastmatch’ ...
** package ‘fastmatch’ successfully unpacked and MD5 sums checked
** libs
gcc -std=gnu99 -I/opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/R/include -DNDEBUG  -I/opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/include    -fpic  -I/opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/include  -c ctapply.c -o ctapply.o
gcc -std=gnu99 -I/opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/R/include -DNDEBUG  -I/opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/include    -fpic  -I/opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/include  -c fasthash.c -o fasthash.o
gcc -std=gnu99 -I/opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/R/include -DNDEBUG  -I/opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/include    -fpic  -I/opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/include  -c fastmatch.c -o fastmatch.o
gcc -std=gnu99 -shared -L/opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/R/lib -L/opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib -lgfortran -o fastmatch.so ctapply.o fasthash.o fastmatch.o -L/opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/R/lib -lR
installing to /opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/R/library/fastmatch/libs
** R
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* creating tarball
packaged installation of ‘fastmatch’ as ‘fastmatch_1.1-0_R_x86_64-pc-linux-gnu.tar.gz’
* DONE (fastmatch)
INFO conda_build.build:bundle_conda(865): Packaging r-fastmatch-1.1_0-r3.3.2_0
number of files: 20
Fixing permissions
patchelf: file: /opt/conda/conda-bld/r-fastmatch_1494873960049/_b_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/R/library/fastmatch/libs/fastmatch.so
    setting rpath to: $ORIGIN/../../../..:$ORIGIN/../../../lib
Fixing permissions
Detected hard-coded path in binary file lib/R/library/fastmatch/libs/fastmatch.so
/opt/conda/conda-bld/linux-64/r-fastmatch-1.1_0-r3.3.2_0.tar.bz2
updating index in: /opt/conda/conda-bld
updating index in: /opt/conda/conda-bld/noarch
updating index in: /opt/conda/conda-bld/linux-64
updating: r-fastmatch-1.1_0-r3.3.2_0.tar.bz2
updating index in: /opt/conda/conda-bld
TEST START: /opt/conda/conda-bld/linux-64/r-fastmatch-1.1_0-r3.3.2_0.tar.bz2
Deleting work directory, /opt/conda/conda-bld/r-fastmatch_1494873960049/work/fastmatch
updating index in: /opt/conda/conda-bld/linux-64
updating index in: /opt/conda/conda-bld/noarch

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    r-fastmatch-1.1_0          |         r3.3.2_0          36 KB  local

The following NEW packages will be INSTALLED:

    bzip2:           1.0.6-1        conda-forge
    ca-certificates: 2017.1.23-1    conda-forge
    cairo:           1.14.6-4       conda-forge
    curl:            7.52.1-0       conda-forge
    fontconfig:      2.12.1-4       conda-forge
    freetype:        2.7-1          conda-forge
    gettext:         0.19.7-1       conda-forge
    glib:            2.51.4-0       conda-forge
    graphite2:       1.3.9-0        conda-forge
    gsl:             2.2.1-1        conda-forge
    harfbuzz:        1.4.3-0        conda-forge
    icu:             58.1-1         conda-forge
    jpeg:            9b-0           conda-forge
    libffi:          3.2.1-3        conda-forge
    libgcc:          5.2.0-0        defaults
    libiconv:        1.14-4         conda-forge
    libpng:          1.6.28-0       conda-forge
    libtiff:         4.0.6-7        conda-forge
    libxml2:         2.9.4-4        conda-forge
    ncurses:         5.9-10         conda-forge
    openssl:         1.0.2k-0       conda-forge
    pango:           1.40.4-0       conda-forge
    pcre:            8.39-0         conda-forge
    pixman:          0.34.0-0       conda-forge
    r-base:          3.3.2-5        conda-forge
    r-fastmatch:     1.1_0-r3.3.2_0 local
    readline:        6.2-0          conda-forge
    tk:              8.5.19-1       conda-forge
    xz:              5.2.2-0        conda-forge
    zlib:            1.2.8-3        conda-forge


WARNING: conda-build appears to be out of date. You have version 2.1.10 but the
latest version is 2.1.13. Run

conda update -n root conda-build

to get the latest version.

+ source /opt/conda/bin/activate /opt/conda/conda-bld/r-fastmatch_1494873960049/_t_env
+ /bin/bash -x -e /opt/conda/conda-bld/r-fastmatch_1494873960049/test_tmp/run_test.sh
+ /opt/conda/conda-bld/r-fastmatch_1494873960049/_t_env/bin/R -e 'library('\''fastmatch'\'')'

R version 3.3.2 (2016-10-31) -- "Sincere Pumpkin Patch"
Copyright (C) 2016 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> library('fastmatch')
>
>
TEST END: /opt/conda/conda-bld/linux-64/r-fastmatch-1.1_0-r3.3.2_0.tar.bz2
# If you want to upload package(s) to anaconda.org later, type:

anaconda upload /opt/conda/conda-bld/linux-64/r-fastmatch-1.1_0-r3.3.2_0.tar.bz2

# To have conda build upload to anaconda.org automatically, use
# $ conda config --set anaconda_upload yes

anaconda_upload is not set.  Not uploading wheels: []



####################################################################################
Source and build intermediates have been left in /opt/conda/conda-bld.
There are currently 1 accumulated.
To remove them, you can run the ```conda build purge``` command
```
**(root) [root@4d77fd2a6b14 bioconda-recipes]# ./simulate-travis.py --disable-docker --package r-fastmatch**
```
+ bioconda-utils lint recipes config.yml --git-range master HEAD
18:47:08 BIOCONDA INFO Recipes newly unblacklisted:

18:47:08 BIOCONDA INFO Recipes to consider according to git:
recipes/r-fastmatch
 recipes/r-fastmatch
18:47:08 BIOCONDA INFO Recipes to lint:
recipes/r-fastmatch
+ set +x
+ bioconda-utils build recipes config.yml --loglevel=info --mulled-test --package r-fastmatch --git-range master HEAD
18:47:09 BIOCONDA INFO Recipes newly unblacklisted:

18:47:09 BIOCONDA INFO Recipes modified according to git: r-fastmatch
18:47:09 BIOCONDA INFO blacklist: bioconductor-bubbletree, bioconductor-cexor, bioconductor-deseq2/1.10.1, bioconductor-diffbind, bioconductor-genelendatabase/1.6.0, bioconductor-iranges/2.4.6, bioconductor-iranges/2.4.7, bioconductor-limma/3.26.7, bioconductor-limma/3.28.2, bioconductor-limma/3.28.6, bioconductor-mmdiff, bioconductor-s4vectors/0.8.7, bioconductor-summarizedexperiment/1.0.2, bioconductor-systempiper, blast/2.2.21, cap-mirseq, denovogear, detonate, gdsctools, mothur/1.36.1, mysqlclient, perl-font-afm, r-hdrcde, r-knitr, r-ks, r-mutoss, r-phonr, r-rainbow, r-readr, r-sartools/1.2.0, triform2
18:47:09 BIOCONDA INFO Filtering recipes

Filtering 1 of 1 (100.0%) recipes/r-fastmatch
recipes/r-fastmatch
Traceback (most recent call last):
  File "/opt/conda/bin/bioconda-utils", line 11, in <module>
    load_entry_point('bioconda-utils==0.10.0', 'console_scripts', 'bioconda-utils')()
  File "/opt/conda/lib/python3.5/site-packages/bioconda_utils/cli.py", line 385, in main
    argh.dispatch_commands([build, dag, dependent, lint])
  File "/opt/conda/lib/python3.5/site-packages/argh/dispatching.py", line 328, in dispatch_commands
    dispatch(parser, *args, **kwargs)
  File "/opt/conda/lib/python3.5/site-packages/argh/dispatching.py", line 174, in dispatch
    for line in lines:
  File "/opt/conda/lib/python3.5/site-packages/argh/dispatching.py", line 277, in _execute_command
    for line in result:
  File "/opt/conda/lib/python3.5/site-packages/argh/dispatching.py", line 260, in _call
    result = function(*positional, **keywords)
  File "/opt/conda/lib/python3.5/site-packages/bioconda_utils/cli.py", line 319, in build
    mulled_upload_target=mulled_upload_target,
  File "/opt/conda/lib/python3.5/site-packages/bioconda_utils/build.py", line 259, in build_recipes
    dag, name2recipes = utils.get_dag(recipes, config=orig_config, blacklist=blacklist)
  File "/opt/conda/lib/python3.5/site-packages/bioconda_utils/utils.py", line 288, in get_dag
    for r in list(load_meta(recipe, config)):
  File "/opt/conda/lib/python3.5/site-packages/bioconda_utils/utils.py", line 108, in load_meta
    yield api.render(recipe, config_obj)[0].meta
  File "/opt/conda/lib/python3.5/site-packages/conda_build/api.py", line 31, in render
    return render_recipe(recipe_path, no_download_source=config.no_download_source, config=config)
  File "/opt/conda/lib/python3.5/site-packages/conda_build/render.py", line 158, in render_recipe
    raise ValueError("no_download_source specified, but can't fully render recipe without"
ValueError: no_download_source specified, but can't fully render recipe without downloading source.  Please fix the recipe, or don't use no_download_source.
```

File: recipes/r-fastmatch/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  number: 0
+
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
+
+requirements:
+  build:
+    - r-base
```
Comment (chambm) [inline recipes/r-fastmatch/meta.yaml]: @bgruening Does this need to be `r` instead of `r-base`?

**Comment by bgruening** [comment]
Your locally is using conda-forge the one in travis is using the `r` channel, can this be a problem?
I think the channel oder from your local setup is different from the bioconda one.

**Comment by chambm** [comment]
Repeating the commands after setting channel priorities didn't change the error:
```
conda config --add channels conda-forge
conda config --add channels defaults
conda config --add channels r
conda config --add channels bioconda
```

**Pull request merged**